From b3137cecfb2551e19946d64a417d0cbe3a511263 Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Tue, 15 Mar 2022 17:02:14 +0100
Subject: [PATCH] Remove the LINK_LANGUAGE genex condition when generating .pc file

Conditions with the LINK_LANGUAGE genex are not parsed correctly when
generating .pc file. So link options will be added unconditionally.

Amends d1e02c385539ce971fbad207ba9aaa32be9c7bac

Fixes: QTBUG-101723
Change-Id: Ib837b3f3429d195a469450ef25af9630ad7d15e2
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 97ef7752a2db871d45c8e434d15e0ccec015cd65)
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
---

diff --git a/cmake/QtPkgConfigHelpers.cmake b/cmake/QtPkgConfigHelpers.cmake
index d88be5f..45bbcbed 100644
--- a/cmake/QtPkgConfigHelpers.cmake
+++ b/cmake/QtPkgConfigHelpers.cmake
@@ -8,6 +8,7 @@
     set(tmpopts "${options}")
     list(FILTER tmpopts EXCLUDE REGEX "\\$<BUILD_INTERFACE:[^,>]+>")
     list(FILTER tmpopts EXCLUDE REGEX "\\$<TARGET_PROPERTY:[^,>]+>")
+    list(TRANSFORM tmpopts REPLACE "\\$<\\$<LINK_LANGUAGE:[^,>]+>:([^,>]+)>" "\\1")
     list(TRANSFORM tmpopts REPLACE "\\$<INSTALL_INTERFACE:([^,>]+)>" "\\1")
     list(TRANSFORM tmpopts REPLACE ">" "$<ANGLE-R>")
     list(TRANSFORM tmpopts REPLACE "," "$<COMMA>")
